Lushu/src/main/kotlin/lushu/ContextGrammar/Grammar/Parser.kt

```diff
@@ -0,0 +1,12 @@
+package lushu.ContextGrammar.Grammar
+
+class Parser(private val rules: Rules = Rules()) {
```

Comment:
```suggestion
class Parser() {
        private val rules: Rules = Rules()
```

I don't see anywhere in the code where we call `Parser` with this parameter, so we can remove it from the parameters and instead put it in the class declaration body.